server/src/server.rs

```diff
@@ -677,32 +677,26 @@ impl Server {
                 if resume_at > now {
                     break;
                 }
-                to_resume.insert((session_id, socket_addr));
+                if let Some(node_session) = server.nodes.get_by_session(session_id.clone()) {
+                    to_resume.push((node_session, session_id, socket_addr));
+                }
                 server.resume_forwarding.remove(elem);
             }
         };
```

Comment:
This way you don't need to clone `server.resume_forwarding`:

```rust
        {
            let now = clock.now();
            let mut server = self.state.write().await;

            // First iteration to release write lock as soon as possible
            for elem in server.resume_forwarding.iter() {
                let (resume_at, session_id, socket_addr) = elem;
                if resume_at > &now {
                    let elem = elem.clone();
                    let mut split = server.resume_forwarding.split_off(&elem);
                    std::mem::swap(&mut split, &mut server.resume_forwarding);
                    break;
                }

                if let Some(node_session) = server.nodes.get_by_session(session_id.clone()) {
                    to_resume.push((node_session, *session_id, *socket_addr));
                }
            }
        };
```

Reply:
Doesn't swap make a copy internally?

Reply:
It doesn't copy as in Rust semantics but in a way that values are moved

Reply:
I would also add a `fixme` to use `drain_filter` instead, when it becomes stable. Alternatively, we could use a different container w/ insertion sort so that the values are still ordered and `drain_filter` is available